lib/rules/selector-id-pattern/index.js

```diff
@@ -32,6 +32,8 @@ const rule = (primary) => {
 		const normalizedPattern = isString(primary) ? new RegExp(primary) : primary;
 
 		root.walkRules((ruleNode) => {
```

Comment:
[ask] How about specifying a selector filter (regex) instead?

```suggestion
		root.walkRules(/#/, (ruleNode) => {
```

Reply:
That's nicer! (Benchmark is comparable.)